cmd/antrea-agent/options.go

```diff
@@ -218,7 +216,11 @@ func (o *Options) validateTLSOptions() error {
 
 func (o *Options) validateAntreaProxyConfig(encapMode config.TrafficEncapModeType) error {
 	if !features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
-		// Validate service CIDR configuration if AntreaProxy is not enabled.
+		o.config.AntreaProxy.Enable = pointer.Bool(false)
```

Comment:
@tnqn are you ok with mutating the config here (in the validation function), or do you think we should add new field under `Options` directly?

Reply:
I feel we should do the latter. Mutating the config data directly may cause confusion when it's later logged or when the whole config is dumpped. At least we haven’t mutated config data (except defaulting) so far AFAIK.

Reply:
@hongliangl it doesn't look like you made that change?

Reply:
Sorry, I forgot this, will address this soon.

Reply:
Done